tachyon/zk/plonk/halo2/prover.h

```diff
@@ -62,6 +66,73 @@ class Prover : public ProverBase<PCS> {
     return ret;
   }
 
+  template <typename Circuit>
+  void CreateProof(
+      const ProvingKey<PCS>& proving_key,
+      std::vector<std::vector<std::vector<F>>>&& instance_columns_vec,
+      std::vector<Circuit>& circuits) {
+    size_t num_circuits = circuits.size();
+
+    // Check length of instances.
+    CHECK_EQ(num_circuits, instance_columns_vec.size());
+    for (const std::vector<std::vector<F>>& instances_vec :
+         instance_columns_vec) {
+      CHECK_EQ(instances_vec.size(), proving_key.verifying_key()
+                                         .constraint_system()
+                                         .num_instance_columns());
+    }
+
+    // Initially write hash value of verification key to transcript.
+    crypto::TranscriptWriter<Commitment>* writer = this->GetWriter();
+    CHECK(writer->WriteToTranscript(
+        proving_key.verifying_key().transcript_repr()));
+
+    // It owns all the columns, polys and the others required in the proof
+    // generation process and provides step-by-step logics as its methods.
+    Argument<PCS> argument =
+        Argument<PCS>::Create(this, circuits, &proving_key.fixed_columns(),
+                              &proving_key.fixed_polys(),
+                              proving_key.verifying_key().constraint_system(),
+                              std::move(instance_columns_vec));
+
+    F theta = writer->SqueezeChallenge();
+    std::vector<std::vector<LookupPermuted<Poly, Evals>>> permuted_lookups_vec =
+        argument.CompressLookupStep(
+            this, proving_key.verifying_key().constraint_system(), theta);
+
+    F beta = writer->SqueezeChallenge();
+    F gamma = writer->SqueezeChallenge();
+    StepReturns<PermutationCommitted<Poly>, LookupCommitted<Poly>,
+                VanishingCommitted<PCS>>
+        committed_result = argument.CommitCircuitStep(
+            this, proving_key.verifying_key().constraint_system(),
+            proving_key.permutation_proving_key(),
+            std::move(permuted_lookups_vec), beta, gamma);
+
+    F y = writer->SqueezeChallenge();
+    argument.TransformAdvice(this->domain());
+    ExtendedEvals circuit_evals = argument.GenerateCircuitPolynomial(
+        this, proving_key, committed_result, beta, gamma, theta, y);
+    VanishingConstructed<PCS> constructed_vanishing;
+    CHECK(CommitFinalHPoly(this, std::move(committed_result).TakeVanishing(),
+                           proving_key.verifying_key(), circuit_evals,
```

Comment:
Sorry but could you rename parameter name of `circuit_column` in `CommitFinalHPoly` to circuit_evals?